src/staff-administrator/components/staff-user-details/staff-user-details.component.ts

```diff
@@ -39,36 +40,55 @@ export class StaffUserDetailsComponent {
   }
 
   public updateUserStatus(): void {
-    if (!this.loading && !this.userDetails.suspended) {
+    if (!this.loading) {
       this.loading = true;
       const staffUser = new StaffUser();
       Object.assign(staffUser, this.userDetails);
-      staffUser.suspended = true;
-      this.staffDataAccessService.updateUser(staffUser).pipe(
-        finalize(() => {
-          this.loading = false;
-          window.scrollTo(0, 0);
-        })
-      )
+      staffUser.suspended = !staffUser.suspended;
+      this.staffDataAccessService.updateUser(staffUser)
         .subscribe(
           () => {
-            this.status = InfoMessageType.SUCCESS;
-            this.title = 'User suspended';
-            this.message = InfoMessage.SUSPEND_USER_SUCCESS;
-            this.userDetails.suspended = staffUser.suspended;
+            this.getUpdatedUser();
```

Comment:
Calling another $.subscribe within the subscribe method is bad practice -- it makes things confusing.


You should be using switchMap 
      this.staffDataAccessService.updateUser(staffUser)
  `this.staffDataAccessService.updateUser(staffUser).pipe(
       finalize(() => {
        this.loading = false;
        window.scrollTo(0, 0);
      }),
      switchMap(() => this.staffDataAccessService.fetchSingleUserById(this.userDetails.case_worker_id)),
      map((user) => this.userDetails = StaffUser.from(user))
    ).subscribe((user) => {
      if (user.up_idam_status !== StaffUserIDAMStatus.PENDING) {
        this.status = InfoMessageType.SUCCESS;
        // User either restored or suspended if successful
        this.title = user.suspended ? 'User suspended' : 'User restored';
        this.message = user.suspended ? InfoMessage.SUSPEND_USER_SUCCESS : InfoMessage.USER_RESTORED;
      }
    },
    (err) => {
      this.setErrorSteps(err);
    });
      
    `